src/kernel/filesystem/fat32.zig

```diff
@@ -0,0 +1,1756 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const expectEqualSlices = std.testing.expectEqualSlices;
+const expectEqual = std.testing.expectEqual;
+const expectError = std.testing.expectError;
+const expect = std.testing.expect;
+const log = std.log.scoped(.fat32);
+const AutoHashMap = std.AutoHashMap;
+const Allocator = std.mem.Allocator;
+const ArrayList = std.ArrayList;
+const vfs = @import("vfs.zig");
+const mem = @import("../mem.zig");
+const CodePage = @import("../code_page.zig").CodePage;
+const mkfat32 = @import("../../../mkfat32.zig");
+
+/// The boot record for FAT32. This is use for parsing the initial boot sector to extract the
+/// relevant information for todays FAT32.
+const BootRecord = struct {
+    /// The jump bytes that begin the boot code. This will jump over the FAT32 header.
+    jmp: [3]u8,
```

Comment:
Do we need to keep this around if the disk is only used for storage and not booting?

Reply:
This is only here for parsing the boot record. This struct isn't used in the main FAT32FS.